packages/manager/modules/telecom-spare/src/modems/modems.controller.js

```diff
@@ -0,0 +1,136 @@
+import map from 'lodash/map';
+
+import modalController from './modal/modems-modal.controller';
+import modalTemplate from './modal/modems-modal.html';
+
+export default class ModemsController {
+  /* @ngInject */
+  constructor(
+    $state,
+    $translate,
+    $uibModal,
+    OvhApiXdsl,
+    TucToast,
+    TucToastError,
+  ) {
+    this.$state = $state;
+    this.$uibModal = $uibModal;
+    this.$translate = $translate;
+    this.OvhApiXdsl = OvhApiXdsl;
+    this.TucToast = TucToast;
+    this.TucToastError = TucToastError;
+  }
+
+  loadRow({ spare }) {
+    return this.OvhApiXdsl
+      .Spare()
+      .v6()
+      .getSpare({ spare })
+      .$promise;
+  }
+
+  orderNewModem() {
+    return this.$state.go('spare.modems.order');
+  }
+
+  refresh() {
+    this.modems = null;
+
+    // reset cache to force reload
+    this.OvhApiXdsl.Spare().v6().resetAllCache();
+
+    return this.OvhApiXdsl.Spare()
+      .v6()
+      .query()
+      .$promise.then((modems) => {
+        this.modems = map(modems, spare => ({ spare }));
+      });
+  }
+
+  /**
+   * Replace modem by its spare
+   */
+  replace({ spare }) {
+    const modal = this.$uibModal.open({
+      animation: true,
+      template: modalTemplate,
+      controller: modalController,
+      controllerAs: '$ctrl',
+      resolve: {
+        params: () => {
+          const params = {
+            action: 'replacement',
+            spare,
+          };
+          return params;
+        },
+      },
+    });
+    modal.result
+      .then(data => this.displayActionResult(data.actionResult))
+      .catch(() => this.displayActionResult(null));
+  }
+
+  /**
+   * Display success or error message in toast message
+   */
+  displayActionResult(result) {
+    if (result) {
+      if (result.isSucceed) {
+        this.TucToast.success(this.$translate.instant(result.messageToDisplay));
+      } else {
+        const error = result.errorMessage;
+        this.TucToast.error(this.$translate.instant(result.messageToDisplay, { error }));
+      }
+      this.refresh();
+    }
+  }
+
+  /**
+   * Delete the spare as if it was not belonging to OVH anymore
+   */
+  delete({ spare }) {
+    const modal = this.$uibModal.open({
```

Comment:
it's better to use ui-router states for modals, you can find some example in the mono-repo. that way the user can refresh the page and directly access the modal (since the modal has a route)

Reply:
I would like to do that but I can't find any example in the mono-repo.Do you have one?

Reply:
https://github.com/ovh-ux/manager/blob/master/packages/manager/modules/pci/src/projects/project/instances/resume/resume.routing.js
You can search through every routing file having a `layout: 'modal'` decalaration to get examples